lib/app.dart

```diff
@@ -31,63 +31,44 @@ class EnteApp extends StatefulWidget {
 
 class _EnteAppState extends State<EnteApp> with WidgetsBindingObserver {
   final _logger = Logger("EnteAppState");
-  final _mediaExtensionPlugin = MediaExtension();
 
   @override
   void initState() {
     _logger.info('init App');
     super.initState();
+    setupIntentAction();
     WidgetsBinding.instance.addObserver(this);
   }
 
-  Future<bool> initIntentAction() async {
-    if (!Platform.isAndroid) {
-      AppLifecycleService.instance.setIntentAction(IntentAction.main);
-      return true;
-    }
-    IntentAction intentAction = IntentAction.main;
-    try {
-      final actionResult = await _mediaExtensionPlugin.getIntentAction();
-      intentAction = actionResult.action!;
-    } on PlatformException {
-      intentAction = IntentAction.main;
-    } catch (error) {
-      _logger.info(error);
-      intentAction = IntentAction.main;
-    }
-    if (intentAction == IntentAction.main) {
+  void setupIntentAction() async {
+    final mediaExtentionAction = await initIntentAction();
```

Comment:
Let's change this to something like this?


`
final mediaExtentionAction = Platform.isAndroid ? await initIntentAction() : MediaExtentionAction(action: Intent.main)
`

Reply:
But by default the value of AppLifeCycleService.Instance.mediaExtensionAction = MediaExtentionAction(action: IntentAction.main);
https://github.com/ente-io/photos-app/pull/876/files#diff-74b0f1f5e15090884b7d3f16a0b37b04e73ac886e6e616c9cbbbba9bb9815d3bR8